pkg/schema/type.go

```diff
@@ -482,6 +484,85 @@ func (n *NullType) SetExamples(exs []Example) {
 	n.documentation.examples = exs
 }
 
+// IsDeprecated provides deprecated field information
+func (t *DocumentType) IsDeprecated() (bool, string) {
+	return false, ""
+}
+
+// IsDeprecated provides deprecated field information
+func (m *MapType) IsDeprecated() (bool, string) {
+	return m.documentation.deprecated, ""
```

Comment:
For each of these lets return the deprecated notice as well, so that consumers of the api can use it.

```suggestion
	return m.documentation.deprecated, a.documentation.deprecationNotice
```

Reply:
Done. Thanks Cari!